### PR TITLE
fix for IDENTITY-3106

### PR DESCRIPTION
**Pull request by dracusds123**
Please review and merge...

### CODE REVIEW

File: components/user-manager/org.wso2.carbon.user.mgt.ui/src/main/java/org/wso2/carbon/user/mgt/ui/UserBean.java [hunk 1]
```diff
@@ -18,46 +18,58 @@
 package org.wso2.carbon.user.mgt.ui;
```
Comment (thariyarox) [inline components/user-manager/org.wso2.carbon.user.mgt.ui/src/main/java/org/wso2/carbon/user/mgt/ui/UserBean.java]: I see lots of places in the diff where no new code modification is done but due to formatting they have come to the diff. Therefore when making PRs in future, please make sure only the code changes you have done are in the PR.

Comment (dracusds123) [inline components/user-manager/org.wso2.carbon.user.mgt.ui/src/main/java/org/wso2/carbon/user/mgt/ui/UserBean.java]: noted. Will PR refactoring separately in the future.

Comment (hpmtissera) [inline components/user-manager/org.wso2.carbon.user.mgt.ui/src/main/java/org/wso2/carbon/user/mgt/ui/UserBean.java]: +1

On Tue, Jun 23, 2015 at 1:24 PM, dracusds123 <notifications@github.com>
wrote:

> In
> components/user-manager/org.wso2.carbon.user.mgt.ui/src/main/java/org/wso2/carbon/user/mgt/ui/UserBean.java
> <https://github.com/wso2/carbon-identity/pull/454#discussion_r33017239>:
>
> > @@ -18,46 +18,58 @@
> >  package org.wso2.carbon.user.mgt.ui;
> >
>
> noted. Will PR refactoring separately in the future.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wso2/carbon-identity/pull/454/files#r33017239>.
>



-- 
Best Regards,
Prasad.

File: components/user-manager/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/add-finish.jsp [hunk 2]
```diff
@@ -55,9 +55,10 @@
             claims = new ClaimValue[]{emailClaim};
             userPassword = null;
         }
-        userBean.addUserRoles((Map<String,Boolean>)session.getAttribute("checkedRolesMap"));
```
Comment (dracusds123) [inline components/user-manager/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/add-finish.jsp]: need this line

**Comment by hpmtissera** [comment]
reviewed

File: components/user-manager/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/add-finish.jsp [hunk 3]
```diff
@@ -55,9 +55,10 @@
...
-        client.addUser(Util.decodeHTMLCharacters(username), userPassword, userBean.getUserRoles(), claims, null);
+        String[] userRoles = userBean.getUserRoles();
+        userBean.removeUserRoles();
```
Comment (dracusds123) [inline components/user-manager/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/add-finish.jsp]: clear this session in 'finally', remove the user bean from session and forward to user-mgt.jsp, remove all session attributes in finally bock

**Pull request closed without merging**